automatic/poi/update.ps1

```diff
@@ -0,0 +1,48 @@
+import-module au
+
+$releases = 'https://github.com/poooi/poi/releases'
+
+function global:au_BeforeUpdate { Get-RemoteFiles -Purge -NoSuffix }
+
+function global:au_SearchReplace {
+    @{
+        ".\tools\chocolateyInstall.ps1" = @{
+            "(?i)(^\s*file\s*=\s*`"[$]toolsDir\\).*" = "`$1$($Latest.FileName32)`""
+        }
+        "$($Latest.PackageName).nuspec" = @{
+            "(\<releaseNotes\>).*?(\</releaseNotes\>)" = "`${1}$($Latest.ReleaseNotes)`$2"
+        }
+    }
+}
+
+function global:au_GetLatest {
+
+    $stableReleaseFound = $false
+    $url = ''
+    $lastVersionString = ''
+    do {
+        $download_page = Invoke-WebRequest -Uri "$releases$lastVersionString" -UseBasicParsing
+        $urls = $download_page.links | ? href -match 'poi-setup-.+\.exe$' | % href
+        $stableUrls = @($urls) -match 'poi-setup-\d+\.\d+\.\d+\.exe$'
+        if($stableUrls.Count -gt 0) {
+            $url = $stableUrls | select -First 1
+            $stableReleaseFound = $true
+        }
+        else {
+            $lastUrl = $urls | select -Last 1
+            $lastVersionString = "?after=$(Split-Path(Split-Path($lastUrl)) -Leaf)"
+        }
+    }
+    until($stableReleaseFound)
```

Comment:
This could potentially loop infinitely, lets say sometime in the future they change how they name the executables. In this case (eventually) this do loop would never exit.

Would be better in to change the match for the links (further up) to something along the lines of:
`$urls $download_page.links | ? href -match 'poi-setup\-d+\.\d+\.\d+\.exe$' | select -first 1 -expand href`
(you could also use the regex `poi-setup\-\d+(\.\d+){1,2}\.exe$`, or just `\d+(\.\d+){1,2}\.exe$`), you'll need to test those first of course.

Reply:
The main concern here is that currently I am only searching for stable versions, and I think it is possible (though unlikely) that the first page of the github release is filled all with beta versions (`1.2.3-beta.0`, `1.2.3-beta.1`, etc), so that no stable versions could be found on the first page, so that one might need to turn to the next page. But I think you are right about the infinite loop, so I might need to set a maximum number of pages to search for (like 3), and if no stable versions can be found in the limit, the script will abort with an error. Do you think that this could solve the problem?

From here, I also have a question about the prerelease versions. Currently I'm not considering beta versions, but how should I do the versioning if I want to include those prerelease versions, when NuGet does not allow versions other than pure numbers separated by dots? Because if that is possible, it would be really easy to parse the github page, since I'll just need to select the latest release, regardless of whether it is a stable or pre release.

Reply:
There are several ways to do this, but the easiest IMO would be to.
1. Add streams to the package, so every version on the page would be checked (see [librecad](https://github.com/chocolatey/chocolatey-coreteampackages/blob/b80c6f0ae6f799818ba0a9aa436fe7dff3faaa9d/automatic/librecad/update.ps1#L34) as an example)
2. Use the `Get-Version` au helper (also used in the librecad updater)

Adding streams is optional though, but would be the best option.
At a minimum though, a change to restrict the amount of looping is needed to be added.

Reply:
Fixed in 155038f. However I still have a small question. (Using the conversation below)